build_tools/benchmarks/collect_compilation_statistics.py

```diff
@@ -152,6 +152,7 @@ def get_module_map_from_compilation_benchmark_config(
           (f"{arch.type.value}-{arch.architecture}-{arch.microarchitecture}-"
            f"{compile_target.target_abi.value}"))
     compilation_info = CompilationInfo(
+        benchmark_name=gen_config.name,
```

Comment:
Wait, shouldn't the CompilationInfo have its own name? The same compilation could be used in multiple benchmark configs, no?

Reply:
Oh, `benchmark_name` here is the name of `CompilationInfo`. I just see it as "compilation benchmark". We can use something like "compilation_name" or just "name" instead.

Reply:
Yeah just "name" seems better to me. I'm still a bit confused about in which cases we use the legacy name. Is `get_module_map_from_benchmark_suite` for deriving from the legacy configs or something? A sprinkling of docstrings would help here :-)

Reply:
Yes, it used in the legacy path `get_module_map_from_benchmark_suite`. Added docs.